lib/rules/no-container.ts

```diff
@@ -0,0 +1,82 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils';
+import { getDocsUrl } from '../utils';
+import {
+  isCallExpression,
+  isIdentifier,
+  isMemberExpression,
+  isObjectPattern,
+  isProperty,
+  isRenderVariableDeclarator,
+} from '../node-utils';
+
+export const RULE_NAME = 'no-container';
+
+export default ESLintUtils.RuleCreator(getDocsUrl)({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow the use of container methods',
+      category: 'Best Practices',
+      recommended: 'error',
+    },
+    messages: {
+      noContainer:
+        'Unexpected use of container methods. Prefer the use of "screen.someMethod()".',
```

Comment:
I'm thinking this rule is more about disabling container, but there's no mention about using screen or not. So perhaps the text could go something like

```suggestion
        'Unexpected use of container methods. Prefer using query methods, such as "getByRole()"',
```

Reply:
Perhaps someone else can come up with a better error text too 😅